src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb

```diff
@@ -114,6 +126,43 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return returnValue
         End Function
 
+        Private Shared Function IsFloatingPointExpressionOfUnknownPrecision(rewrittenNode As BoundExpression) As Boolean
+            If rewrittenNode Is Nothing Then
+                Return False
+            End If
+
+            ' Note: no special handling for node having a constant value because it cannot reach here
+
+            Dim type = rewrittenNode.Type
```

Comment:
Minor point: Consider setting the local value to `rewrittenNode.Type.SpecialType` instead.